app/src/main/java/ayds/winchester/songinfo/moredetails/model/repository/external/ExternalRepositoryImpl.kt

```diff
@@ -0,0 +1,25 @@
+package ayds.winchester.songinfo.moredetails.model.repository.external
+
+import ayds.winchester.songinfo.moredetails.model.entities.Description
+import retrofit2.Response
+import java.io.IOException
+
+interface ExternalRepository{
+    fun getArtistDescription(artistName: String): Description
+}
+
+internal class ExternalRepositoryImpl (private var wikipediaAPI : WikipediaAPI, private var wikipediaToDescriptionResolver : WikipediaToDescriptionResolver) : ExternalRepository{
+
+    private lateinit var queryWikipediaSearch : Response<String>
+    private lateinit var artistName : String
+
+     override fun getArtistDescription(artistName: String): Description{
+        this.artistName = artistName
+        try {
+            queryWikipediaSearch = wikipediaAPI.getArtistInfo(artistName).execute()
+        } catch (e1: IOException) {
+            e1.printStackTrace()
+        }
+        return wikipediaToDescriptionResolver.getDescriptionFromExternalData(queryWikipediaSearch)
+    }
```

Comment:
No son necesarias las propiedades. Ademas, si ocurre una excepción en el try catch, la linea 23 tirará una excepción "late init var not initialized". No hace falta el try catch acá, ya que lo hace el repository
```suggestion

     override fun getArtistDescription(artistName: String): Description{
         val queryWikipediaSearch = wikipediaAPI.getArtistInfo(artistName).execute()
         return wikipediaToDescriptionResolver.getDescriptionFromExternalData(queryWikipediaSearch)
    }
```